tables/TaQL/ExprNodeArray.cc

```diff
@@ -66,47 +66,47 @@ TENShPtr TableExprNodeArray::makeConstantScalar()
       {
         MArray<Bool> arr = getArrayBool(0);
         if (arr.size() == 1) {
-          return new TableExprNodeConstBool (arr.array().data()[0]);
+          return TENShPtr(new TableExprNodeConstBool (arr.array().data()[0]));
```

Comment:
Optional, also for the `return` statements in the other `case`s.
```suggestion
          return std::make_shared<TableExprNodeConstBool>(arr.array().data()[0]);
```

Reply:
done